qiita_db/meta_util.py

```diff
@@ -184,12 +184,16 @@ def update_redis_stats():
     number_of_samples = {}
     ebi_samples_prep = {}
     num_samples_ebi = 0
+
     for k, sts in viewitems(studies):
         number_of_samples[k] = 0
         for s in sts:
             st = s.sample_template
             if st is not None:
-                number_of_samples[k] += len(list(st.keys()))
+                for p in s.prep_templates():
```

Comment:
Note that this will ignore all studies/samples without preps/artifacts; which I think it's fine but just FYI.

Reply:
Correct, but I also think it's fine. I wasn't sure how to handle the case where there isn't an artifact because visibility is denoted at the artifact, not the template level, right?

Reply:
Good question!! Historically, status was given per study, then we moved to do it by artifact, then we realized that admins will need to click on each artifact to make all artifacts private/public so we decided to do this per prep. However, the status is kept by artifact and just propagated to all artifacts in a given preparation.